datastream-server-restli/src/test/java/com/linkedin/datastream/server/dms/TestDatastreamResources.java

```diff
@@ -47,14 +53,15 @@
 import com.linkedin.restli.server.PathKeys;
 import com.linkedin.restli.server.RestLiServiceException;
 
-
 /**
  * Test DatastreamResources with ZooKeeper backed DatastreamStore
  */
 @Test(singleThreaded = true)
 public class TestDatastreamResources {
```

Comment:
Some of the unit-tests are missing:

a) Trying to stop the datastream in STOPPING state.
b) Trying to pause/resume the datastream in STOPPING state.
c) Trying to delete the datastream in STOPPING state.
d) Leader busy and not able to pick the LEADER_DO_ASSIGNMENT message or busy updating the zookeeper, and the request timed-out.

Reply:
I have added tests a, b & c. Any suggestions on how can we test the (d) scenario in unit tests?

Reply:
for d) You can mock the leader coordinator, mark `fetchDatastreamGroupsWithStatus` `visibleForTesting` and return empty list when this api is called with STOPPING status.

Another thing you can do is extract a method out of fetchDatastreamGroupsWithStatus(Collections.singletonList(DatastreamStatus.STOPPING)); and return empty list.

Reply:
It was harder to mock the coordinator, so mocked the state store to achieve this scenario.